Added gRPC-ATS

abahetik opened the pull request:
Adding gRPC ATS

ni-sujain commented:
CMakeLists.txt file has some steps to for code generation and project creation for test folder. Since we are moving the test folder to deprecated I think we need to either update the CMakeLists.txt or comment out that thing for now, we can file a github issue to fix that later.

abahetik commented:
In this PR, I have added gRPC-ATS. Things that has been done in the PR are:

1. I have moved the previous Tests to a Deprecated folder. This tests will be fixed later.
2. run_ci_tests.yml - I have added a new workflow for running ci tests
3. Testing.md - Added a new md file on How to run and add tests.
4. I have added the Test Infrastructure. It includes:

   1. Tests.lst : This file consists of the Path to all the TestVIs.
   2. AutoTests - This folder should contain all the TestVIs.
   3. run_tests.py - This file runs all the tests in Tests.lst through the LabVIEW CLI using the gRPC_ATS.vi
   4. gRPC_ATS.vi - This is wrapper around LV_ATS.vi. This is run using the LabVIEW CLI.
   5. LV_ATS.vi - This VI runs the Test VIs and gathers information of its success or failure
![image](https://user-images.githubusercontent.com/64770378/201631064-15ab6cd3-d8ae-41e3-9835-00eecc00c602.png)

   7. TestUtility folder - It has the error ctls understood by the ATS. These ctls are supposed to be used by the TestVIs.

5. Added an example test(HelloWorldTest.vi)
![image](https://user-images.githubusercontent.com/64770378/201633367-8e6812e2-1449-4c4f-8685-2f3fcda8fdc4.png)

ni-sujain commented:
Why we have a change in greeter_server.vi from hello world example, can you post a diff
